client/post-editor/index.js

```diff
@@ -14,6 +14,7 @@ module.exports = function() {
 	page( '/post', controller.pressThis, sitesController.siteSelection, sitesController.sites );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page( '/post/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.post );
+	page( '/post/revision/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.revision );
```

Comment:
Things are less clear in the rest of Calypso because of (some) support for manipulating multiple sites at once, but in the post editor the URLs follow the convention of `/verb/site/more-specific-things-here`.  Viewing `revisions` is a more specific action than viewing a particular post ID, so it should come after the post ID in the URL.

I think the only way adding an extra URL position in `getSiteFragment` would break anything is if we have a URL like `/verb.with.dots/site.wordpress.com/numeric/numeric` (something that would be newly and incorrectly recognized as a site ID in the `length - 3` position).  This seems extremely unlikely to me.

In the past when changing a function like `getSiteFragment`, I've successfully used a testing approach like this:

- Implement both versions of the function side-by-side
- Add debugging to print out return values from both versions of the code, especially if they differ
- Navigate around the app, hitting as many sections as possible

Unfortunately for the issues you pointed to, this was missed in some of our longer and harder-to-reach routes (#9542 and #9550 for specific examples).

I agree that using `page.js` params would be a better design, but it's a much larger (and to me, much scarier) change - currently `getSiteFragment` is called using an actual parameterized URL rather than a `page.js` route, so this would require restructuring all of the code where it is used.

Finally I agree with @youknowriad's suggestion that we keep this PR to the Redux changes only.  There are other things to discuss around URLs, like what the possible flows look like for getting to this feature (different post types, etc) and what that implies about the possible URL structures.